saas/api/transactions.py

```diff
@@ -390,6 +392,52 @@ def list(self, request, *args, **kwargs):
                 " processor (ie. %s)." % str(err)})
 
 
+class ImportTransactionsAPIView(ProviderMixin, APIView):
+    """
+    Insert transactions that were done offline for the purpose of computing
+    accurate metrics.
+
+    """
+
+    def post(self, request, *args, **kwargs):
+        form = ImportTransactionForm(request.data)
+        if form.is_valid():
```

Comment:
Take a look at [rest_framework/mixins.py](https://github.com/encode/django-rest-framework/blob/90ed2c1ef7c652ce0a98075c422d25a632d62507/rest_framework/mixins.py#L18) to see how code should be written. Override `perform_create` to fit with the DRF library patterns.